src/collect/collect.c

```diff
@@ -484,10 +489,12 @@ int main(int argc, char **argv)
         kickout();
 
         lseek(fd, 0, SEEK_SET);
-        ftruncate(fd, 0);
+        if (ftruncate(fd, 0)) { /* TODO: implement proper error handling */
```

Comment:
ditto; print and fail here should be fairly easy to add

Reply:
OK, thanks!

I am also extending the check to `lseek`